src/jit/codegencommon.cpp

```diff
@@ -10611,6 +10612,17 @@ void CodeGen::genPreserveCalleeSavedFltRegs(unsigned lclFrameSize)
         regMaskTP regBit = genRegMask(reg);
         if ((regBit & regMask) != 0)
         {
+#ifdef FEATURE_AVX_SUPPORT
+            // when we reach here, function does not contain AVX instruction so far, however, since copyIns can
+            // be an AVX instruction such as vmovupd, we should check and issue vzeroupper before the copyIns to
+            // avoid Legacy SSE code (from native code such as Reverse PInvoke) to AVX transition penalty
+            if (compiler->getFloatingPointInstructionSet() == InstructionSet_AVX && !bVzeroupperIssued &&
+                getEmitter()->IsAVXInstruction(copyIns))
+            {
+                instGen(INS_vzeroupper);
+                bVzeroupperIssued = true;
+            }
+#endif
```

Comment:
Fixed, given lower has set the AVX flags already, this can be removed. Also will change genVzeroupperIfNeeded() return value to void since we don't need to check the return value any more.